.github/workflows/container-image.yml

```diff
@@ -39,6 +42,26 @@ jobs:
           push: true
           tags: |
             ghcr.io/rancher-sandbox/lockc:latest
+      -
+        name: Retrieve pull request number
+        if: ${{ startsWith(github.ref, 'refs/pull/') }}
```

Comment:
@nickgerace I have an explanation for those `if` statements now.

So, this whole file is triggered by many events which are mutual exclusive when it comes to pushing images. We trigger this workflow on:
* push to `main` branch (merging a PR) (`github.ref` is `refs/heads/*`)
* creating a tag (`github.ref` is `refs/tags/*`)
* submitting a pull request (`github.ref` is `refs/pull/*`)

For all of those actions we want to build an image, but we want to push it differeently. To handle then differently, we need to catch them with if statements.

That's why we do:
* `if: ${{ startsWith(github.ref, 'refs/pull/') }}`
* `if: ${{ startsWith(github.ref, 'refs/heads/') }}`
* `if: ${{ startsWith(github.ref, 'refs/tags/') }}`

Reply:
thanks for the explanation!